highlight/events-explainer.md

```diff
@@ -66,21 +67,11 @@ Here's an example illustrating how a "find-highlights" HighlightRangeGroup could
 </html>
 ```
 
-### [Multiple Ranges and HighlightRangeGroups](#multiple-ranges-and-highlightrangegroups)
+### [Multiple Ranges and Highlights](#multiple-ranges-and-highlights)
 
-Ranges can overlap and each range may belong to more than one HighlightRangeGroup. If more than one Range or HighlightRangeGroup are hit, there are different options for disambiguating which HighlightRangeGroups receive the pointer event, as well as how many such events the target group should receive.
+Ranges can overlap and each range may belong to more than one Highlight. However, only one event will be fired on one Highlight. Highlights have a priority order as defined in the [Highlight API explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/highlight/explainer.md). To find the group that will receive the event, sort the Highlight objects in descending priority order and dispatch the event to the first Highlight. The priority order establishes a layering of highlights, so firing the event based on this ordering ensures the highlight "nearest to the user" receives the event.

```

Comment:
There are two distinct cases here and I think we should call them out separately.

**Case 1**: There are multiple overlapping ranges, each belonging to separate `Highlight`s. What happens when a user clicks where the ranges overlap?

**Case 2**: There is a single range, which has been added to multiple `Highlight`s. What happens when a user clicks the range?

Case 1 seems like it could happen in a lot of common scenarios, e.g. find-on-page finds a misspelled word, and the user clicks it. For a lot of these cases I think it's reasonable to only fire one event in this case, e.g. the author probably doesn't need both find-on-page behavior and spell-check behavior to happen when the user clicks the word.

Maybe there are other cases where the author would want both `Highlight`s to receive events though. If a word is highlighted for both a grammar error and a spelling error, the author may want to include some logic in their event handlers to coordinate dynamically whether the grammar or spelling error gets to handle a click, rather than having it always go to whichever `Highlight` has higher priority.

We should spend some time thinking through the different use cases here to try to make a case for which approach will be better for authors.

I'm having a harder time reasoning about Case 2. It seems suspicious for an author to use a single range in multiple `Highlight`s. Perhaps here we should just do whatever is simpler. Although either firing on all `Highlight`s vs firing on only the highest pri seem equally simple to me; either way we have to define a priority of highlights.

We don't have to answer all these questions yet in this explainer but I think it is worth mentioning both scenarios.